banyand/property/shard.go

```diff
@@ -225,3 +228,33 @@ func (s *shard) search(ctx context.Context, indexQuery index.Query, limit int,
 	}
 	return data, nil
 }
+
+func (s *shard) prepareForMerge(src []*roaring.Bitmap, segments []segment.Segment, _ uint64) (dest []*roaring.Bitmap, err error) {
+	if len(segments) == 0 || len(src) == 0 || len(segments) != len(src) {
+		return src, nil
+	}
+	for segID, seg := range segments {
+		var docID uint64
+		for ; docID < seg.Count(); docID++ {
+			hasDeleted := false
+			_ = seg.VisitStoredFields(docID, func(field string, value []byte) bool {
+				if field == deleteField {
+					hasDeleted = convert.BytesToBool(value)
+					return false
+				}
+				return true
+			})
+
+			if !hasDeleted {
```

Comment:
You should set a flag to control whether the data is removed after a fixed duration, with a default duration of 7 days.